src/OpenTelemetry.Resources.AssemblyMetadata/README.md

```diff
@@ -0,0 +1,137 @@
+# Assembly Metadata Resource Detector
+
+| Status      |                                                    |
+|-------------|----------------------------------------------------|
+| Stability   | [Alpha](../../README.md#alpha)                     |
+| Code Owners | [@austindrenski](https://github.com/austindrenski) |
+
+[![NuGet version badge](https://img.shields.io/nuget/v/OpenTelemetry.Resources.AssemblyMetadata)](https://www.nuget.org/packages/OpenTelemetry.Resources.AssemblyMetadata)
+[![NuGet download count badge](https://img.shields.io/nuget/dt/OpenTelemetry.Resources.AssemblyMetadata)](https://www.nuget.org/packages/OpenTelemetry.Resources.AssemblyMetadata)
+[![codecov.io](https://codecov.io/gh/open-telemetry/opentelemetry-dotnet-contrib/branch/main/graphs/badge.svg?flag=unittests-Resources.AssemblyMetadata)](https://app.codecov.io/gh/open-telemetry/opentelemetry-dotnet-contrib?flags[0]=unittests-Resources.AssemblyMetadata)
+
+## Getting Started
+
+You need to install the `OpenTelemetry.Resources.AssemblyMetadata` package to be able to use the Assembly Metadata Resource Detector.
+
+```shell
+dotnet add package --prerelease OpenTelemetry.Resources.AssemblyMetadata
+```
+
+## Usage
+
+You can add the Assembly Metadata Resource Detector to the `ResourceBuilder` with the following example:
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Resources;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddOpenTelemetry()
+    .ConfigureResource(static builder => builder.AddAssemblyMetadataDetector())
+    .WithLogging(static builder => /* ... */)
+    .WithMetrics(static builder => /* ... */)
+    .WithTracing(static builder => /* ... */);
+```
+
+The resource detector will record attributes defined on the [entry assembly](https://learn.microsoft.com/dotnet/api/system.reflection.assembly.getentryassembly) using [AssemblyMetadataAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblymetadataattribute) with keys prefixed by `otel:`.
```

Comment:
This would also need to be updated if the overload for specifying the assemblies were added.

```suggestion
The resource detector will add resource attributes defined on the [entry assembly](https://learn.microsoft.com/dotnet/api/system.reflection.assembly.getentryassembly) using [AssemblyMetadataAttribute](https://learn.microsoft.com/dotnet/api/system.reflection.assemblymetadataattribute) with keys prefixed by `otel:`.
```